### DISCUSSION TITLE
Overview for Database

### DISCUSSION THREAD

**Discussion by rebio** [lensfun/lensfun#1614] [Ideas]
Hello there!

I was looking for a list of all supported lenses and cameras so that I can check if a) the lenses I own are supported and b) I can add my lenses to the database.

- For people searching for a database overview: https://wilson.bronger.org/lensfun_coverage.html

The database stored in _lensfun/data/db/_ is of course not designed to be human legible. So finding the right lens or camera takes a good amount of time (eg. I searched quite a bit before I found out that smartphones are stored in _misc.xml_).

My questions would be:

1. Is there already a database in lensfun and I just missed it
2. If there is not, what is your thought about creating a HTML overview like wilson.bronger.org, but that is build into the project? (Or at least to add a link to the list on wilson.bronger.org right from a readme file in the database folder)

**Comment by sarunasb** [comment]
A good place to do this would be repository of Lensfun website https://github.com/lensfun/lensfun.github.io

**Reply by Sturmflut** [reply]
I had a look at it a while ago. GitHub pages can't do the XML processing on its own, so we would need something external that pulls lensfun master, converts the XML files into an HTML overview, commits the HTML files to the lensfun.github.io repository and pushes the commits.

Basically what @bronger has been doing on wilson.bronger.org, but with a more complex step (push the end result to a GitHub repository instead of to a webserver directory) at the end.

I guess it could be automated with GitHub Actions and secrets.

**Reply by seebk** [reply]
Wouldn't it be possible to let Github Actions do the whole job of generating and pushing the new lens list? 

Some time ago we had Travis CI set up to recompile the website when a change was detected to the sources and then it pushed the new files to the https://github.com/lensfun/lensfun.github.io repo. Now Lensfun uses GitHub Actions so this update does not work anymore, but I suspect this can be fixed somehow to work the same way with GitHub Actions.

**Reply by seebk** [reply]
Here's a link to the old Travis CI config, maybe that helps...

https://github.com/lensfun/lensfun/blob/ed9b250ccbef6e4f439f6b44c64cb9d3d942bdc0/.travis.yml

**Reply by Sturmflut** [reply]
I've just had a look at it. The Travis config only pushes the API documentation, it doesn't update the list of supported lenses. The lists for older releases are static markdown files, the list for the master branch is a link to [http://wilson.bronger.org/lensfun_coverage.html](http://wilson.bronger.org/lensfun_coverage.html).

**Reply by seebk** [reply]
Yes, but there also is a Python script in the Lensfun `tools/lenslist` folder which can automatically generate the list of lenses. One can choose between HTML or Markdown output of the list by commandline flags. 

https://github.com/lensfun/lensfun/blob/master/tools/lenslist/show_lensfun_coverage.py

I have always used this script to generate the static Markdown/Jekyll lists for the Lensfun releases you can see on the website. Basically that's the same script used to generate the list for http://wilson.bronger.org/lensfun_coverage.html as well.

I think it should be possible to trigger a GitHub action by a commit to the `db/` subfolder, generate a Markdown list of the current lenses in the database with the Python script and then push this to the Lensfun website repo. Not sure about the details though ;)

**Reply by sarunasb** [reply]
Github action within lensfun/lensfun works fine (#1662). Not sure how to get it accross to website repo.

**Reply by seebk** [reply]
I had a little spare time this weekend and have updated the website and main repo GitHub Actions to automatically create and push the develop lens list on the website. Will try to do the same for the API docs soon.
